### PR TITLE
fix(rpc): encode timestamp as string in BlockHeaderView

### PR DESCRIPTION
**Pull request by bowenwang1996**
Bridge testing is broken because of timestamp being too large (larger than u53) and therefore we cannot dump headers properly.

### CODE REVIEW

File: core/primitives/src/views.rs [hunk 1]
```diff
@@ -328,6 +328,7 @@ pub struct BlockHeaderView {
     pub outcome_root: CryptoHash,
     pub chunks_included: u64,
     pub challenges_root: CryptoHash,
+    #[serde(with = "u64_dec_format")]
```
Comment (bowenwang1996) [inline core/primitives/src/views.rs:331]: @frol should I create a separate field `real_timestamp` for backward compatibility?

**Comment by ailisp** [comment]
Yes, will also need to correct timestamp in inner light
 http post https://rpc.testnet.near.org/ jsonrpc=2.0 method=next_light_client_block params:='["D2MW6a7tNiLjVXsfioMAKDboxP4s6tQVSZNHuRJ4umYY"]' id="dontcare"

**Comment by bowenwang1996** [comment]
@nearmax they were always in nanoseconds. The issue is not with borsh serialization. Rather it is because we obtain the light client block from rpc and parses it as json. The reason that some test succeeds is because the undefined behavior of json if the number exceeds u53.

**Comment by ailisp** [comment]
> @nearmax they were always in nanoseconds. The issue is not with borsh serialization. Rather it is because we obtain the light client block from rpc and parses it as json. The reason that some test succeeds is because the undefined behavior of json if the number exceeds u53.

nearbridge/test/block_121498.json works seems too lucky to explain, it ends with `000` (seems to lost precision), but it pass the test, means the block is exactly produce at an exact milisecond?

**Comment by bowenwang1996** [comment]
> nearbridge/test/block_121498.json works seems too lucky to explain, it ends with 000 (seems to lost precision), but it pass the test, means the block is exactly produce at an exact milisecond?

That is possible. Maybe it's also related to the operating system.

**Comment by nearmax** [comment]
@ailisp We would also need to modify this code here: https://github.com/near/rainbow-bridge/blob/37e21e8f29a992b99b41b6074312341f5d924160/environment/lib/near2eth-relay.js#L29 to first cast `block.inner_lite.timestamp` from string to u64 and then borshify it.

**Comment by ailisp** [comment]
> That is possible. Maybe it's also related to the operating system.

You're right, from docs.rs:
> Each Duration is composed of a whole number of seconds and a fractional part represented in nanoseconds. If the underlying system does not support nanosecond-level precision, APIs binding a system timeout will typically round up the number of nanoseconds.

**Comment by bowenwang1996** [comment]
> You're right, from docs.rs:

I am surprised at my ability to randomly guess things :)

**Comment by nearmax** [comment]
@ailisp FYI I have updated bridge to work with this change and `executor_id` in https://github.com/near/rainbow-bridge/pull/130 .

**Comment by bowenwang1996** [comment]
@frol looks like this got merged accidentally. I guess we have to submit a separate PR if you have comments :(

**Comment by ailisp** [comment]
@bowenwang1996 Looks good to me, going to deploy a new devnet to dump headers

**Comment by frol** [comment]
@bowenwang1996 Yeah, we should have made a new field and leave the current field to be "JSON number" even though it is overflowed if parsed on JS side. Can you make a PR with that? I suggest we also name the new field as `timestamp_nanosec`.

**Comment by bowenwang1996** [comment]
@frol sure

**Pull request merged**